upgrades/upgrade_testing_framework/robot_lib/OpenSearchRESTActions.py

```diff
@@ -1,34 +1,44 @@
 from pathlib import Path
 import json
 from typing import Any
-
-from opensearchpy import OpenSearch
+import cluster_migration_core.core.versions_engine as ev
+import cluster_migration_core.clients as clients
 
 DEFAULT_TEMP_STORAGE = "/tmp/utf/robot_data.json"
 
 
-# Change this library to use REST requests instead of Python client
-class OpenSearchClientLibrary(object):
+class CouldNotRetreiveAttributeFromRESTResponse(Exception):
+    def __init__(self, key: str, response: dict):
+        super().__init__(f"Could not find key '{key}' in the RESTResponse json: {response}")
+
+
+class OpenSearchRESTActions(object):
 
     def __init__(self, host="localhost", port=9200, temp_storage_location=DEFAULT_TEMP_STORAGE):
-        os_client = OpenSearch(hosts=[{'host': host, 'port': port}])
-        self._os_client = os_client
+        #This is temporary, this will be provided by the executor
+        engine_version = ev.EngineVersion(ev.ENGINE_OPENSEARCH, 1, 3, 6)
```

Comment:
Hmm, is there a reason we didn't pipe this through in this PR?  We're using the same set of actions against both ES 7.10.2 and OS 1.3.6 Clusters in our current workflow, right?  This feels a bit too hacky to me.

Reply:
@chelma I didn't want to bog Omar down with routing the EngineVersion from the steps to the actual actions here and have a PR to raise which will remove this.

@okhasawn I think I'd be a little more comfortable with removing the engine version lines here and just using  `self._rest_client = RESTClientDefault()` until my change goes in and reworks this